pallets/manta-pay/src/lib.rs

```diff
@@ -1184,68 +1253,65 @@ where
     fn is_valid(
         &self,
         posting_key: TransferPostingKeyRef<config::Config, Self>,
-    ) -> Result<Option<(Self::ValidProof, Self::Event)>, TransferLedgerError> {
+    ) -> Result<(Self::ValidProof, Self::Event), TransferLedgerError<T>> {
         let transfer_shape = TransferShape::from_posting_key_ref(&posting_key);
         if transfer_shape.is_none() {
-            return Ok(None);
+            return Err(TransferLedgerError::InvalidTransferShape);
         }
```

Comment:
I think we can delete this check and turn below `expect()` into a `.ok_or(TransferLedgerError::InvalidTransferShape)?`

Reply:
That'd be nice yeah